extended-it/src/test/java/apoc/vectordb/WeaviateEnterpriseTest.java

```diff
@@ -0,0 +1,128 @@
+                .waitingFor(Wait.forLogMessage(".*Started..*\\n", 1))
+                .withNeo4jConfig("dbms.transaction.timeout", "60s");
+        neo4jContainer.start();
+        session = neo4jContainer.getSession();
+
+
+        WEAVIATE_CONTAINER.waitingFor(Wait.forLogMessage(".*Serving weaviate at http://.*\\n", 1));
+
+        WEAVIATE_CONTAINER.start();
+        HOST = "http://host.docker.internal:" + WEAVIATE_CONTAINER.getMappedPort(WEAVIATE_PORT);
+
+        testCall(session, WEAVIATE_CREATE_COLLECTION_APOC,
+                map("host", HOST, "conf", ADMIN_HEADER_CONF),
+                r -> {
+                    Map value = (Map) r.get("value");
+                    assertEquals("TestCollection", value.get("class"));
+                });
+
+        testResult(session, WEAVIATE_UPSERT_QUERY,
+                map("host", HOST, "id1", ID_1, "id2", ID_2, "conf", ADMIN_HEADER_CONF),
+                r -> {
+                    Map<String, Object> row = r.next();
+                    Map<String, Object> value = (Map<String, Object>) row.get("value");
+
+                    assertEquals(COLLECTION_NAME, value.get("class"));
+
+                    while (r.hasNext()) {
+                        row = r.next();
+                        value = (Map<String, Object>) row.get("value");
+                        assertEquals(COLLECTION_NAME, value.get("class"));
+                    }
+                    assertFalse(r.hasNext());
+                });
+
+        // -- delete vector
+        testCall(session, WEAVIATE_DELETE_VECTOR_APOC,
+                map("host", HOST, "conf", ADMIN_HEADER_CONF),
+                r -> {
+                    List value = (List) r.get("value");
+                    assertEquals(List.of("7ef2b3a7-1e56-4ddd-b8c3-2ca8901ce308", "7ef2b3a7-1e56-4ddd-b8c3-2ca8901ce309"), value);
+                });
+    }
+
+    @AfterClass
+    public static void tearDown() throws Exception {
+        testCallEmpty(session, WEAVIATE_DELETE_COLLECTION_APOC,
+                MapUtil.map("host", HOST, "collectionName", COLLECTION_NAME, "conf", ADMIN_HEADER_CONF)
+        );
+        session.close();
+        neo4jContainer.close();
+        WEAVIATE_CONTAINER.stop();
+    }
+
+    @Test
+    public void queryVectors() {
+        testResult(session,  WEAVIATE_QUERY_APOC,
+                map("host", HOST, "conf", map(ALL_RESULTS_KEY, true, FIELDS_KEY, FIELDS, HEADERS_KEY, ADMIN_AUTHORIZATION)),
+                r -> {
+                    Map<String, Object> row = r.next();
+                    assertBerlinResult(row, ID_1, FALSE);
+                    assertNotNull(row.get("score"));
+                    assertNotNull(row.get("vector"));
+
+                    row = r.next();
+                    assertLondonResult(row, ID_2, FALSE);
+                    assertNotNull(row.get("score"));
+                    assertNotNull(row.get("vector"));
```

Comment:
Mettiamoci pure questo nello util insieme all'altro di WeaviateTest, 
chiamandolo `public static void queryVectorsAssertions(Result r)`

Reply:
Purtroppo il testResult di WeaviateEnterpriseTest restituisce un Iterator<Map<String, Object>> e non un Result come in WeaviateTest

Reply:
Ora faccio qualche test perché potrebbe funzionare se il Result eredita qualche Iterator